rest_framework/versioning.py

```diff
@@ -127,6 +132,17 @@ def reverse(self, viewname, args=None, kwargs=None, request=None, format=None, *
             viewname, args, kwargs, request, format, **extra
         )
 
+    def resolve(self, path, urlconf=None, request=None):
+        match = django_resolve(path, urlconf)
+        if match.namespace:
```

Comment:
Rather than returning a new match is it possible to just mutate the one we've gotten back?
Eg.

    match.namespaces = match.namespaces[1:]